### PR TITLE
fix(escape-special-characters): escape special characters in user routes (DSP-1557)

### PR DESCRIPTION
**Pull request by irinaschubert**
resolves DSP-1557

### CODE REVIEW

File: webapi/src/main/scala/org/knora/webapi/responders/admin/UsersResponderADM.scala [hunk 1]
```diff
@@ -1504,6 +1439,78 @@ class UsersResponderADM(responderData: ResponderData) extends Responder(responde
...
+    * @throws BadRequestException         if necessary parameters are not supplied.
+    * @throws UpdateNotPerformedException if the update was not performed.
+    */
+  private def updateUserPasswordADM(userIri: IRI,
+                                    password: Password,
+                                    featureFactoryConfig: FeatureFactoryConfig,
+                                    requestingUser: UserADM,
+                                    apiRequestID: UUID): Future[UserOperationResponseADM] = {
+
+    log.debug("updateUserPasswordADM - new password: {}", password.value)
```
Comment (BalduinLandolt) [inline webapi/src/main/scala/org/knora/webapi/responders/admin/UsersResponderADM.scala:1460]: isn't it a bit risky to log the password like this? In a real life scenario, a user's password would end up somewhere in the docker logs.

**Review by BalduinLandolt — APPROVED** [review]
looks good to me

File: webapi/src/test/scala/org/knora/webapi/responders/admin/UsersResponderADMSpec.scala [hunk 2]
```diff
@@ -606,22 +513,23 @@ class UsersResponderADMSpec extends CoreSpec(UsersResponderADMSpec.config) with
         /* Password is updated by other normal user */
         responderManager ! UserChangePasswordRequestADM(
           userIri = SharedTestDataADM.superUser.id,
-          changeUserRequest = ChangeUserApiRequestADM(
-            requesterPassword = Some("test"),
-            newPassword = Some("test123456")
+          userUpdatePasswordPayload = UserUpdatePasswordPayloadADM(
+            requesterPassword = Password.create("test").fold(error => throw error, value => value),
```
[the remark this answers is not in the document]
Comment (irinaschubert) [reply webapi/src/test/scala/org/knora/webapi/responders/admin/UsersResponderADMSpec.scala:517]: I absolutely agree.

File: webapi/src/main/scala/org/knora/webapi/responders/admin/UsersResponderADM.scala (hunk 1 above)
Comment (irinaschubert) [reply webapi/src/main/scala/org/knora/webapi/responders/admin/UsersResponderADM.scala:1460]: Absolutely, this is not good at all. It this was there before, I deleted this line now.

File: webapi/src/main/twirl/org/knora/webapi/messages/twirl/queries/sparql/admin/updateUserPassword.scala.txt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+}
+
+@*
+
+GraphDB's consistency checking requires reasoning, but reasoning interferes with certain things
+in the WHERE clauses of our SPARQL updates, so we set a GraphDB-specific flag to return only
+explicit statements in the WHERE clause here.
+
+*@
```
[the remark this answers is not in the document]
Comment (irinaschubert) [reply webapi/src/main/twirl/org/knora/webapi/messages/twirl/queries/sparql/admin/updateUserPassword.scala.txt:67]: I have no idea... As there are more places with graph-db specific details and comments, I suggest leaving it here and make an own task with removing all remnants at once. I added a task for it to our Miro board.

File: webapi/src/test/scala/org/knora/webapi/messages/admin/responder/valueObjects/ValueObjectsADMSpec.scala [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+import org.knora.webapi.messages.StringFormatter
+import org.knora.webapi.messages.admin.responder.usersmessages._
+import org.knora.webapi.{IRI, UnitSpec}
+import org.scalatest.enablers.Messaging.messagingNatureOfThrowable
+
+object ValueObjectsADMSpec {
+  val config: Config = ConfigFactory.parseString("""
+          akka.loglevel = "DEBUG"
+          akka.stdout-loglevel = "DEBUG"
+        """.stripMargin)
```
[the remark this answers is not in the document]
Comment (irinaschubert) [reply webapi/src/test/scala/org/knora/webapi/messages/admin/responder/valueObjects/ValueObjectsADMSpec.scala:33]: Unfortunately, multiline strings are not handled consistently at all. Actually, this config block appears in a lot of test files without pipes (although there are some cases where pipes are used). I added it now to the YT issue regarding reformatting, so we can decide on how we deal with multiline strings.

File: webapi/src/test/scala/org/knora/webapi/e2e/admin/UsersADME2ESpec.scala [hunk 5]
```diff
@@ -405,25 +401,79 @@ class UsersADME2ESpec
...
 
-      "create the user if the supplied email is unique " in {
+        response.status should be(StatusCodes.OK)
+
+        val result: UserADM = AkkaHttpUtils.httpResponseToJson(response).fields("user").convertTo[UserADM]
+        result.givenName should be("Updated\tGivenName")
+        result.familyName should be("Updated\"FamilyName")
+      }
+
+      "escape special characters when getting the user" in {} //TODO continue here
```
[the remark this answers is not in the document]
Comment (irinaschubert) [reply webapi/src/test/scala/org/knora/webapi/e2e/admin/UsersADME2ESpec.scala:458]: No, I totally forgot about this and added the test now. Thanks for spotting this!

**Pull request merged**